definitions/features/sync_plans.rb

```diff
@@ -33,12 +33,24 @@ def sync_plan_ids_by_status(enabled = true, filter_ids = nil)
     feature(:foreman_database).query(query).map { |r| r['id'].to_i }
   end
 
+  def validate_sync_plan_ids(ids)
+    ids_condition = ids.map { |id| "'#{id}'" }.join(',')
+    query = <<-SQL
+      SELECT id FROM katello_sync_plans WHERE id IN (#{ids_condition})
+    SQL
+    feature(:foreman_database).query(query).map { |r| r['id'].to_i }
+  end
```

Comment:
Can we do this without querying the database directly? Either through hammer or the API?

Reply:
Would be possible but in the same file, there are other SQL querries. Therefore I thought, this is the way to implement this.

Reply:
Yea... I was hoping we could start to not make that problem worse.

Reply:
Well, currently there is a bug which should be solved. Receiving the still existing sync plan ids via hammer wouldn't be pretty performant. 

In another PR, I'm pretty sure the required_new_implementation can bei removed.  Maybe then its a good point in time to make some parts easier.

Reply:
Welp, you can't blame me for trying to stem the tide of bad design in the project. Thanks for at least considering it.

Reply:
This was surely not my intention. The question is, should we first improve the design of sync-plan feature and afterwards fix the issue (a), fix the bug first and then improve the design (b) or do it in one PR (c)?

Reply:
I am mostly worried doing it systematically will be hard, so where we can, try not to introduce new instances of database access. That does not mean you have to solve it here to solve the real bug, not as a follow up (unless you really really want to). I just have the one logic question -- https://github.com/theforeman/foreman_maintain/pull/681/files#r1112268541 and then we can proceed with merging